### ISSUE TITLE
power up nothing on ui display

### ISSUE THREAD

**Issue by mciucci65** [tebl/RC-ONE#3]
I finish to assemble the RC1 but when powerup nothing is display on UI.

I'm using 28c256 need to do something to prepare bin file for programming ?

the 5v power it's ok led on keypaad is onand i alse checked with multimeter.
I did some more test and if I remove the riot chip 6532 some was displayed on UI.

not for last I'm using full working MOS 6502A with jumper VP enabled.

Can anyone help me ?
I'm no sure that 6532 is full working china chip :-(

**Comment by tebl** [comment]
It's been quite a while since I looked at a 6502, so the following may be a
bit more vague than what I would've wanted it to be.

The 6532 may be a likely culprit as any chip from China is likely to have
been pulled from a different system, and at times they get rebadged
something else - simply by having the same overall appearance. That's not a
guarantee though as you'd need to ensure that the 6502 is properly running
on its own, pay particular attention to the NMI and RESET signals - using a
multimeter, ensure that they momentarily go low when pushed and then go
high again. The clock is harder to observe without an oscilloscope, but a
multimeter will average over time so you'll see a voltage somewhere in the
middle (without being stuck high or low). Also make sure to follow the
steps in the Troubleshooting-document.

The board was designed for the 28C256, so just ensure that the jumpers are
set to where you put the firmware (or write it to each 8k-parts of the
chip, or just do what I do in the batch files and combine the data four
times to get 32k worth of data). A good guide to start reading for the
hardware-side, is this one: http://wilsonminesco.com/6502primer/.

If you do end up getting nowhere, one tip is to look at basic examples for
the assembly code. That way you can have the computer do exactly one thing,
and then try to observe it doing that specific thing. For example, just
doing a NOP-loop, then have it write a single value to a specific address
in RAM.

lør. 9. nov. 2024 kl. 17:14 skrev mciucci65 ***@***.***>:

**Comment by mciucci65** [comment]
thanks for quick answer it's ok if I use a MOS 6502A CPU ?
Sure cpu it's working because a tested it on other 6502 project.

I checked with oscilloscope the clock on pin 37 and 39 of cpu it's ok. The reset signal it's ok I can see down/up signal.
Can I test it with rs232 connection ? If yes give me the parameter for serial connection.

Other question the firmware 28c64 in firmware folder on this github it's ready to use? can I use it for duplicate for 28c256 eeprom ?

kind regards
MC

**Comment by tebl** [comment]
Yes, a 6502A should work fine - as far as I know the only differences
should be its capability to run at speeds up to 2Mhz. The files in the
folder is what I've used on my computer, you can just look at the batch
file to see how you can combine four copies of it to fill the entire 28C256
(always helps to have one less thing to think about).

søn. 10. nov. 2024 kl. 05:43 skrev mciucci65 ***@***.***>:

**Comment by mciucci65** [comment]
Hi thanks again,

The file for eeprom it's ok.  At this point I think that the problem is 6532 RIOT so a I ask the refund and took one from another china seller :-)...

I see on UI PCB that there is I think a serial port, Is there no way to check via the serial port if you see anything? And if so, what are the parameters to use?

With MOS 6502A the VP jumper must be jumpered ? 

kind regards
MC

**Comment by tebl** [comment]
Serial is a bit harder as it's software-driven, so it's the first thing to
break if there's a problem somewhere. Easiest way to test is probably to
write a small program that alternates the outputs to the display.

tir. 12. nov. 2024, 11:01 skrev mciucci65 ***@***.***>:

**Comment by mciucci65** [comment]
Hi Tebl,

I'm waiting for new 6532 but now I have a doubt:

for eeprom 28c256 I must do four copies of 28c64.bin or I need to compose the eeprom in diffrente mode:

I tried:

1. copy /b "28c64.bin"+"28c64.bin"+"28c64.bin"+"28c64.bin" "28c256.bin"

2. copy /b "blank_2k.bin"+"blank_2k.bin"+"blank_2k.bin"+"blank_2k.bin"+"blank_2k.bin"+"blank_2k.bin"+"blank_2k.bin"+"blank_2k.bin"+"blank_2k.bin"+"blank_2k.bin"+"blank_2k.bin"+"blank_2k.bin"+"blank_2k.bin"+"blank_2k.bin"+"blank_2k.bin"+"system.bin" "28c256.bin"

3. copy /b "blank_2k.bin"+"blank_2k.bin"+"blank_2k.bin"+"system.bin"+"blank_2k.bin"+"blank_2k.bin"+"blank_2k.bin"+"blank_2k.bin"+"blank_2k.bin"+"blank_2k.bin"+"blank_2k.bin"+"blank_2k.bin"+"blank_2k.bin"+"blank_2k.bin"+"blank_2k.bin"+"blank_2k.bin" "28c256.bin"

but nothing with all eeprom bin files

kind regards

**Comment by tebl** [comment]
Number 1) should work regardless of what the A13/A14 jumpers are set to, as
all 8k slots are the same.

Number 2) would only work with both  jumpers set to high.

Number 3) would only work with both jumpers set to low.

Also check that DEN, EXT_RAM, EXT_ROM has been jumpered correctly.

tor. 14. nov. 2024, 12:56 skrev mciucci65 ***@***.***>:

**Comment by mciucci65** [comment]
thank you for your great support the issue has been resolved after I changed the 6532 received today.
thank again
kind regards
mc

**Comment by tebl** [comment]
Glad it worked! Nothing more satisfying than a well-earned victory over the
machine 😅

tir. 19. nov. 2024, 13:22 skrev mciucci65 ***@***.***>:
